app/database/database_setup.py

```diff
@@ -0,0 +1,124 @@
+import sqlite3
+
+from app.util.conf.configuration import Configuration
+
+
+class FailedSQLiteTableSetup(Exception):
+    pass
+
+
+class DatabaseSetup:
+    _builds_query = """CREATE TABLE IF NOT EXISTS builds (
+        build_id INTEGER PRIMARY KEY,
+        completed BOOLEAN
+    )"""
+
+    _build_metas_query = """CREATE TABLE IF NOT EXISTS build_metas (
+        build_id INTEGER PRIMARY KEY,
+        artifacts_tar_file TEXT,
+        artifacts_zip_file TEXT,
+        error_message TEXT,
+        postbuild_tasks_are_finished TEXT,
+        setup_failures INTEGER,
+        timing_file_path TEXT
+    )"""
+
+    _build_artifacts_query = """CREATE TABLE IF NOT EXISTS build_artifacts (
+        build_id INTEGER,
+        build_artifact_dir TEXT
+    )"""
+
+    _failed_artifact_directories_query = """CREATE TABLE IF NOT EXISTS failed_artifact_directories (
+        uid INTEGER PRIMARY KEY,
+        build_id INTEGER,
+        failed_artifact_directory TEXT
+    )"""
+
+    _failed_subjob_atom_pairs_query = """CREATE TABLE IF NOT EXISTS failed_subjobs_atom_pairs (
+        uid INTEGER PRIMARY KEY,
+        build_id INTEGER,
+        subjob_id INTEGER,
+        atom_id INTEGER
+    )"""
+
+    _build_requests_query = """CREATE TABLE IF NOT EXISTS build_requests (
+        build_id INTEGER PRIMARY KEY,
+        build_parameters STRING
+    )"""
+
+    _build_fsm_query = """CREATE TABLE IF NOT EXISTS build_fsms (
+        build_id INTEGER PRIMARY KEY,
+        state TEXT,
+        queued REAL,
+        finished REAL,
+        prepared REAL,
+        preparing REAL,
+        error REAL,
+        canceled REAL,
+        building REAL
+    )"""
+
+    _subjobs_query = """CREATE TABLE IF NOT EXISTS subjobs (
+        uid INTEGER PRIMARY KEY,
+        subjob_id INTEGER,
+        completed BOOLEAN,
+        build_id INTEGER
+    )"""
+
+    _atoms_query = """CREATE TABLE IF NOT EXISTS atoms (
+        uid INTEGER PRIMARY KEY,
+        atom_id INTEGER,
+        build_id INTEGER,
+        subjob_id INTEGER,
+        command_string TEXT,
+        expected_time REAL,
+        actual_time REAL,
+        exit_code INTEGER,
+        state TEXT
```

Comment:
Is there a reason to do explicit create table statements like this instead of doing a `Base.metadata.create_all` call?